test/test_sparse.py

```diff
@@ -2177,6 +2177,55 @@ def test_factory_empty_indices(self):
         expected_indices = torch.empty((4, 0), dtype=torch.long, device=device)
         self.assertEqual(tensor._indices(), expected_indices)
 
+    def test_sparse_eye(self):
+        device = 'cuda' if self.is_cuda else 'cpu'
+        data_type = [torch.long, torch.int, torch.float, torch.half, torch.double, torch.short, torch.float64]
+
+        for data_type in (torch.long, torch.int, torch.float, torch.half, torch.double, torch.short, torch.float64):
+            size = 4
+            expected_indices = torch.arange(size, device=device, dtype=torch.long).unsqueeze(0).expand(2, -1)
```

Comment:
In addition to putting the tests below into this loop, compare torch.eye with a sparse layout with torch.eye with a strided layout by calling to_dense on the sparse result. Also assert that the sparse tensor returned from eye is coalesced with `is_coalesced`.

The current assertEquals that compare indices and values separately can be removed.

Reply:
@mruberry - Updated the test case as per your suggestion, kindly review the same